cwl/src/main/scala/cwl/WorkflowStepInputExpression.scala

```diff
@@ -23,18 +24,65 @@ final case class WorkflowStepInputExpression(input: WorkflowStepInput,
 
   override def evaluateValue(inputValues: Map[String, WomValue], ioFunctionSet: IoFunctionSet) = {
 
-    def lookupValue(key: String): Checked[WomValue] =
+    def lookupValue(key: String): ErrorOr[WomValue] =
       inputValues.
         get(FullyQualifiedName(key).id).
-        toRight(s"source value $key not found in input values ${inputValues.mkString("\n")}.  Graph Inputs were ${graphInputs.mkString("\n")}" |> NonEmptyList.one)
+        toValidNel(s"source value $key not found in input values ${inputValues.mkString("\n")}.  Graph Inputs were ${graphInputs.mkString("\n")}")
 
-    (input.valueFrom, input.source.map(_.fold(WorkflowStepInputSourceToStrings))) match {
-      case (None, Some(List(id))) =>
-        lookupValue(id).toValidated
+    (input.valueFrom, input.source.map(_.fold(StringOrStringArrayToStringList)), input.effectiveLinkMerge) match {
+
+      //When we have a single source, simply look it up
+      case (None, Some(List(source)), LinkMergeMethod.MergeNested) =>
+        lookupValue(source)
+
+      //When we have several sources, validate they are all present and provide them as a nested array of maps
```

Comment:
Can you explain briefly why in this case we create `WomMap`s ?

Reply:
Hm.  I recall thinking that this is getting around the sticky situation if they're different types, but this is indeed wrong!  I'm looking at a conformance test \#25 that is expecting just the value, not a map.  Thanks for catching this!